app/src/main/java/org/fossasia/susi/ai/data/DeviceModel.kt

```diff
@@ -26,16 +30,21 @@ class DeviceModel : IDeviceModel {
                     listener.onSendCredentialFailure(t.localizedMessage)
             }
 
-            override fun onResponse(call: Call<String>?, response: Response<String>?) {
+            override fun onResponse(call: Call<SpeakerWifiResponse>, response: Response<SpeakerWifiResponse>) {
                 listener.onSendCredentialSuccess()
             }
         })
     }
 
-    override fun setConfiguration(stt: String, tts: String, hotword: String, wake: String, listener: IDeviceModel.onSetConfigurationListener) {
+    override fun setConfiguration(speakerConfig: SpeakerConfiguration, listener: IDeviceModel.onSetConfigurationListener) {
+        val query: MutableMap<String, String> = HashMap()
+        query.put("stt", speakerConfig.stt)
+        query.put("tts", speakerConfig.tts)
+        query.put("hotword", speakerConfig.hotword)
+        query.put("wake", speakerConfig.wake)
 
```

Comment:
You can make a static method in ClientBuilder.java where you can create this map there and return a call from there.

Reply:
ClientBuilder is only for api calls to susi server, but these calls are made to the local server of the device.

Reply:
@iamareebjamal there is no clientbuilder for device service should this be changed?

Reply:
Ideally there should be a wrapper over retrofit services which does this stuff and validation, processing before return etc

But previous work which @arundhati24 worked on was already built and used, so it was a huge change to be done there. This is new implementation, add a service on top which internally calls the API